compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -9758,7 +9758,9 @@ private void handleSafeNavigation(BLangBlockStmt blockStmt, BLangAccessExpressio
             if (this.safeNavigationAssignment != null) {
                 this.safeNavigationAssignment.expr = addConversionExprIfRequired(accessExpr, tempResultVar.getBType());
             }
-            return;
+            if (Types.getReferredType(originalType).tag != TypeTags.JSON) {
```

Comment:
```ballerina
import ballerina/io;

type FooRec1 record {
    json j;
};

type FooRec3 record {
    FooRec1 fooRec?;
};

public function main() returns error? {
    FooRec3 rec2 = {fooRec: {j: {k: "3", l: {m: "4"}}}};
    json val = check rec2.fooRec?.j.k;

    io:println(val); 
}
```
I believe the problem lies with the type checker. In the example mentioned, the type of the expression rec2.fooRec?.j is set as 'json', It should be 'json|()' instead.

Reply:
`json` already has `()`, so we shouldn't have to use `json|()`.